tests/ContextTest.php

```diff
@@ -31,65 +31,9 @@ public function testCanBeValidatedIndependently()
         $this->assertEquals([], $result->getMessages());
     }
 
-    public function testCanHaveIndependentMessages()
-    {
-        $this->validator->context('insert', function (Validator $context) {
-            $context->required('first_name')->length(5);
-
-            $context->overwriteMessages([
-                'first_name' => [
-                    Rule\Length::TOO_SHORT => 'This is from inside the context.'
-                ]
-            ]);
-        });
-
-        $this->validator->overwriteMessages([
-            'first_name' => [
-                Rule\Length::TOO_SHORT => 'This is outside of the context',
-            ]
-        ]);
-
-        $result = $this->validator->validate(['first_name' => 'Rick'], 'insert');
-        $expected = [
-            'first_name' => [
-                Rule\Length::TOO_SHORT => 'This is from inside the context.'
-            ]
-        ];
-        $this->assertEquals($expected, $result->getMessages());
-    }
-
-    public function testMessagesWillBeInheritedFromDefaultContext()
```

Comment:
This is no longer a use-case, since the messages will always be overwritten from the validation result. This decreases the complexity, which is another plus of this feature.